added option to clip input files for questionable files

sleep-yearning opened the pull request:
default behavior doesn't change, but if the clip parameter is passed as True it's passed on to the mido backend which already implements the same behavior.

This should resolve #164 
Edit: might increase the version number for mido requirement

sleep-yearning committed:
added option to clip input files for questionable files

default behavior doesn't change, but if the clip parameter is passed as True it's passed on to the mido backend which already implements the same behavior.

sleep-yearning commented:
After looking through the mido versions, the required version should be 1.2.8, that's when clipping got implemented there.
Idk if that's the reason for the Travis Check fail though.

sleep-yearning committed:
updated required mido version to work with clipping function

sleep-yearning committed:
formatting to pass build checks

craffel commented:
Hey, thanks for this. To be honest I don't think that corrupt MIDI files should be read in this manner - it seems dangerous to arbitrarily clip any non-127 data byte to 127. If a `pretty_midi` user wanted this behavior, I'd suggest they load in the MIDI file with mido with `clip=True`, write it out to a tempfile, and then read it in with `pretty_midi`. I could be convinced otherwise. Either way, you'd need to add the `clip` argument to the docstring.

sleep-yearning commented:
Alright, I get where you're coming from. An explanation why it was useful for me: 
I wanted to use pretty-midi to analyze and convert large sets of midi files. 
Having files in there which stop the whole process is pretty cumbersome, especially if you can be tolerant for _some_ values being off.

I switched from mido to pretty-midi in my code and had to change it anyway for me to work.
And since another user requested this too, I added the PR. 
Let me know if you want to pull or not, I will add the docstring then.

craffel commented:
In that case I would suggest wrapping your MIDI file loading in a try/except and skipping the problematic files. If a data byte is > 127 I would treat the file as corrupt, i.e. not useful.
